### ISSUE TITLE
Java error - failed to write core dump

### ISSUE THREAD

**Issue by pengxiao78** [bcbio/bcbio-nextgen#2242]
Hi Brad,

When I run the tumor-normal somatic calling pipeline, I found the following error.  I have used "ulimit -c unlimited" in the slurm since long time ago and the current bcbio_nextgen and GATK enviroment have been used and been stable for a long time. However, this error just came out recently. Could you please let me for debugging?

Thanks!

Peng

```
[2018-01-30T15:14Z] icb4: Unexpected error
Traceback (most recent call last):
  File "/path/to/parallel/bcbio/anaconda/lib/python2.7/site-packages/bcbio/distributed/ipythontasks.py", line 50, in _setup_logging
    yield config
  File "/path/to/parallel/bcbio/anaconda/lib/python2.7/site-packages/bcbio/distributed/ipythontasks.py", line 344, in variantcall_sample
    return ipython.zip_args(apply(genotype.variantcall_sample, *args))
  File "/path/to/parallel/bcbio/anaconda/lib/python2.7/site-packages/bcbio/variation/genotype.py", line 316, in variantcall_sample
    call_file = caller_fn(align_bams, items, sam_ref, assoc_files, region, call_file)
  File "/path/to/parallel/bcbio/anaconda/lib/python2.7/site-packages/bcbio/variation/gatk.py", line 129, in haplotype_caller
    broad_runner.run_gatk(params)
  File "/path/to/parallel/bcbio/anaconda/lib/python2.7/site-packages/bcbio/broad/__init__.py", line 313, in run_gatk
    log_error=log_error)
  File "/path/to/parallel/bcbio/anaconda/lib/python2.7/site-packages/bcbio/provenance/do.py", line 22, in run
    _do_run(cmd, checks, log_stdout, env=env)
  File "/path/to/parallel/bcbio/anaconda/lib/python2.7/site-packages/bcbio/provenance/do.py", line 102, in _do_run
    raise subprocess.CalledProcessError(exitcode, error_msg)
CalledProcessError: Command 'set -o pipefail; java -Xms681m -Xmx5454m -XX:+UseSerialGC -Djava.io.tmpdir=/path/to/batch4/work/bcbiotx/tmpGZ8oPz -jar /path/to/parallel/bcbio/toolplus/gatk/3.4-46-gbc02625/GenomeAnalysisTK.jar -R /path/to/parallel/bcbio/genomes/Hsapiens/GRCh37/seq/GRCh37.fa --annotation BaseQualityRankSumTest --annotation FisherStrand --annotation GCContent --annotation HaplotypeScore --annotation HomopolymerRun --annotation MappingQualityRankSumTest --annotation MappingQualityZero --annotation QualByDepth --annotation ReadPosRankSumTest --annotation RMSMappingQuality --annotation DepthPerAlleleBySample --annotation Coverage -I /path/to/batch4/work/bamprep/WG036/GL000216.1/WG036-sort-GL000216.1_0_172294-prep.bam --dbsnp /path/to/parallel/bcbio/genomes/Hsapiens/GRCh37/variation/dbsnp-147.vcf.gz -L /path/to/batch4/work/gatk-haplotype/GL000216.1/WG036-germline-GL000216.1_0_172294-regions-nolcr.bed --interval_set_rule INTERSECTION -T HaplotypeCaller -o /path/to/batch4/work/bcbiotx/tmpiUMz8l/WG036-germline-GL000216.1_0_172294.vcf.gz --annotation ClippingRankSumTest --annotation DepthPerSampleHC --pair_hmm_implementation VECTOR_LOGLESS_CACHING -ploidy 2 -U LENIENT_VCF_PROCESSING --read_filter BadCigar --read_filter NotPrimaryAlignment
INFO  09:14:40,263 HelpFormatter - --------------------------------------------------------------------------------- 
INFO  09:14:40,265 HelpFormatter - The Genome Analysis Toolkit (GATK) v3.4-46-gbc02625, Compiled 2015/07/09 17:38:12 
INFO  09:14:40,265 HelpFormatter - Copyright (c) 2010 The Broad Institute 
INFO  09:14:40,265 HelpFormatter - For support and documentation go to http://www.broadinstitute.org/gatk 
INFO  09:14:40,269 HelpFormatter - Program Args: -R /path/to/parallel/bcbio/genomes/Hsapiens/GRCh37/seq/GRCh37.fa --annotation BaseQualityRankSumTest --annotation FisherStrand --annotation GCContent --annotation HaplotypeScore --annotation HomopolymerRun --annotation MappingQualityRankSumTest --annotation MappingQualityZero --annotation QualByDepth --annotation ReadPosRankSumTest --annotation RMSMappingQuality --annotation DepthPerAlleleBySample --annotation Coverage -I /path/to/batch4/work/bamprep/WG036/GL000216.1/WG036-sort-GL000216.1_0_172294-prep.bam --dbsnp /path/to/parallel/bcbio/genomes/Hsapiens/GRCh37/variation/dbsnp-147.vcf.gz -L /path/to/batch4/work/gatk-haplotype/GL000216.1/WG036-germline-GL000216.1_0_172294-regions-nolcr.bed --interval_set_rule INTERSECTION -T HaplotypeCaller -o /path/to/batch4/work/bcbiotx/tmpiUMz8l/WG036-germline-GL000216.1_0_172294.vcf.gz --annotation ClippingRankSumTest --annotation DepthPerSampleHC --pair_hmm_implementation VECTOR_LOGLESS_CACHING -ploidy 2 -U LENIENT_VCF_PROCESSING --read_filter BadCigar --read_filter NotPrimaryAlignment 
INFO  09:14:40,273 HelpFormatter - Executing as peng.xiao@icb4 on Linux 3.10.0-693.5.2.el7.x86_64 amd64; OpenJDK 64-Bit Server VM 1.7.0_161-mockbuild_2017_12_06_14_27-b00. 
INFO  09:14:40,273 HelpFormatter - Date/Time: 2018/01/30 09:14:40 
INFO  09:14:40,273 HelpFormatter - --------------------------------------------------------------------------------- 
INFO  09:14:40,274 HelpFormatter - --------------------------------------------------------------------------------- 
INFO  09:14:40,443 GenomeAnalysisEngine - Strictness is SILENT 
INFO  09:14:40,555 GenomeAnalysisEngine - Downsampling Settings: Method: BY_SAMPLE, Target Coverage: 500 
INFO  09:14:40,563 SAMDataSource$SAMReaders - Initializing SAMRecords in serial 
INFO  09:14:40,635 SAMDataSource$SAMReaders - Done initializing BAM readers: total time 0.07 
INFO  09:14:40,646 HCMappingQualityFilter - Filtering out reads with MAPQ < 20 
INFO  09:14:40,815 IntervalUtils - Processing 101304 bp from intervals 
WARN  09:14:40,819 IndexDictionaryUtils - Track dbsnp doesn't have a sequence dictionary built in, skipping dictionary validation 
INFO  09:14:40,877 GenomeAnalysisEngine - Preparing for traversal over 1 BAM files 
INFO  09:14:40,893 GenomeAnalysisEngine - Done preparing for traversal 
INFO  09:14:40,893 ProgressMeter - [INITIALIZATION COMPLETE; STARTING PROCESSING] 
INFO  09:14:40,893 ProgressMeter -                 |      processed |    time |         per 1M |           |   total | remaining 
INFO  09:14:40,894 ProgressMeter -        Location | active regions | elapsed | active regions | completed | runtime |   runtime 
INFO  09:14:40,894 HaplotypeCaller - Disabling physical phasing, which is supported only for reference-model confidence output 
INFO  09:14:40,983 StrandBiasTest - SAM/BAM data was found. Attempting to use read data to calculate strand bias annotations values. 
INFO  09:14:40,983 StrandBiasTest - SAM/BAM data was found. Attempting to use read data to calculate strand bias annotations values. 
INFO  09:14:41,005 HaplotypeCaller - Using global mismapping rate of 45 => -4.5 in log10 likelihood units 
Using AVX accelerated implementation of PairHMM
#
# A fatal error has been detected by the Java Runtime Environment:
#
#  SIGSEGV (0xb) at pc=0x00007f001b6a2ce9, pid=165395, tid=139639172101888
#
# JRE version: OpenJDK Runtime Environment (7.0_161) (build 1.7.0_161-mockbuild_2017_12_06_14_27-b00)
# Java VM: OpenJDK 64-Bit Server VM (24.161-b00 mixed mode linux-amd64 compressed oops)
# Derivative: IcedTea 2.6.12
# Distribution: CentOS Linux release 7.4.1708 (Core) , package rhel-2.6.12.0.el7_4-x86_64 u161-b00
# Problematic frame:
# C  [libVectorLoglessPairHMM5103467785081113790.so+0x1bce9]  LoadTimeInitializer::LoadTimeInitializer()+0x1669
#
# Failed to write core dump. Core dumps have been disabled. To enable core dumping, try "ulimit -c unlimited" before starting Java again
#
# An error report file with more information is saved as:
# /tmp/jvm-165395/hs_error.log
#
# If you would like to submit a bug report, please include
# instructions on how to reproduce the bug and visit:
#   http://icedtea.classpath.org/bugzilla
#
/usr/bin/bash: line 1: 165395 Aborted                 (core dumped) java -Xms681m -Xmx5454m -XX:+UseSerialGC -Djava.io.tmpdir=/path/to/batch4/work/bcbiotx/tmpGZ8oPz -jar /path/to/parallel/bcbio/toolplus/gatk/3.4-46-gbc02625/GenomeAnalysisTK.jar -R /path/to/parallel/bcbio/genomes/Hsapiens/GRCh37/seq/GRCh37.fa --annotation BaseQualityRankSumTest --annotation FisherStrand --annotation GCContent --annotation HaplotypeScore --annotation HomopolymerRun --annotation MappingQualityRankSumTest --annotation MappingQualityZero --annotation QualByDepth --annotation ReadPosRankSumTest --annotation RMSMappingQuality --annotation DepthPerAlleleBySample --annotation Coverage -I /path/to/batch4/work/bamprep/WG036/GL000216.1/WG036-sort-GL000216.1_0_172294-prep.bam --dbsnp /path/to/parallel/bcbio/genomes/Hsapiens/GRCh37/variation/dbsnp-147.vcf.gz -L /path/to/batch4/work/gatk-haplotype/GL000216.1/WG036-germline-GL000216.1_0_172294-regions-nolcr.bed --interval_set_rule INTERSECTION -T HaplotypeCaller -o /path/to/batch4/work/bcbiotx/tmpiUMz8l/WG036-germline-GL000216.1_0_172294.vcf.gz --annotation ClippingRankSumTest --annotation DepthPerSampleHC --pair_hmm_implementation VECTOR_LOGLESS_CACHING -ploidy 2 -U LENIENT_VCF_PROCESSING --read_filter BadCigar --read_filter NotPrimaryAlignment
' returned non-zero exit status 134
```

**Comment by chapmanb** [comment]
Peng;
Thanks much for the report. Is this a consistent problem if you re-run the pipeline or does it only happen sporadically? Sometimes java just segfaults, so re-running is the best way to fix. If you're finding it happening consistently on the same file then we could try to debug further. Hope this helps.

**Comment by pengxiao78** [comment]
Brad,
This is a consistent problem. I have tried many times.
Thanks!

**Comment by rdocking** [comment]
For what it's worth, I just saw a similar error using the latest development as of this morning:

```
Using GATK jar /projects/rdocking_prj/software/bcbio-nextgen/data/anaconda/share/gatk4-4.0.0.0-0/gatk-package-4.0.0.0-local.jar
Running:
    java -Dsamjdk.use_async_io_read_samtools=false -Dsamjdk.use_async_io_write_samtools=true -Dsamjdk.use_async_io_write_tribble=false -Dsamjdk.compression_level=1 -Xms727m -Xmx3272m -XX:+UseSerialGC -Djava.io.tmpdir=/projects/karsanscratch/rdocking/KARSANBIO-1290_holiday_setup/A34430_small/work/joint/gatk-haplotype-joint/NA12878_small_batch/chr1_gl000191_random/bcbiotx/tmptqq5o0 -jar /projects/rdocking_prj/software/bcbio-nextgen/data/anaconda/share/gatk4-4.0.0.0-0/gatk-package-4.0.0.0-local.jar GenomicsDBImport --reader-threads 1 --genomicsdb-workspace-path NA12878_small_batch-chr1_gl000191_random_0_106433_genomicsdb -L chr1_gl000191_random:1-106433 --variant /projects/karsanscratch/rdocking/KARSANBIO-1290_holiday_setup/A34430_small/work/variation/rnaseq/gatk-haplotype/NA12878_small-gatk-haplotype.vcf.gz
12:10:33.872 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/projects/rdocking_prj/software/bcbio-nextgen/data/anaconda/share/gatk4-4.0.0.0-0/gatk-package-4.0.0.0-local.jar!/com/intel/gkl/native/libgkl_compression.so
12:10:33.974 INFO  GenomicsDBImport - ------------------------------------------------------------
12:10:33.974 INFO  GenomicsDBImport - The Genome Analysis Toolkit (GATK) v4.0.0.0
12:10:33.974 INFO  GenomicsDBImport - For support and documentation go to https://software.broadinstitute.org/gatk/
12:10:33.975 INFO  GenomicsDBImport - Executing as rdocking@clingen01 on Linux v2.6.32-504.8.1.el6.x86_64 amd64
12:10:33.975 INFO  GenomicsDBImport - Java runtime: OpenJDK 64-Bit Server VM v1.8.0_121-b15
12:10:33.975 INFO  GenomicsDBImport - Start Date/Time: January 31, 2018 12:10:33 PM PST
12:10:33.975 INFO  GenomicsDBImport - ------------------------------------------------------------
12:10:33.975 INFO  GenomicsDBImport - ------------------------------------------------------------
12:10:33.975 INFO  GenomicsDBImport - HTSJDK Version: 2.13.2
12:10:33.975 INFO  GenomicsDBImport - Picard Version: 2.17.2
12:10:33.975 INFO  GenomicsDBImport - HTSJDK Defaults.COMPRESSION_LEVEL : 1
12:10:33.975 INFO  GenomicsDBImport - HTSJDK Defaults.USE_ASYNC_IO_READ_FOR_SAMTOOLS : false
12:10:33.975 INFO  GenomicsDBImport - HTSJDK Defaults.USE_ASYNC_IO_WRITE_FOR_SAMTOOLS : true
12:10:33.975 INFO  GenomicsDBImport - HTSJDK Defaults.USE_ASYNC_IO_WRITE_FOR_TRIBBLE : false
12:10:33.976 INFO  GenomicsDBImport - Deflater: IntelDeflater
12:10:33.976 INFO  GenomicsDBImport - Inflater: IntelInflater
12:10:33.976 INFO  GenomicsDBImport - GCS max retries/reopens: 20
12:10:33.976 INFO  GenomicsDBImport - Using google-cloud-java patch 6d11bef1c81f885c26b2b56c8616b7a705171e4f from https://github.com/droazen/google-cloud-java/tree/dr_all_nio_fixes
12:10:33.976 INFO  GenomicsDBImport - Initializing engine
12:10:34.297 INFO  IntervalArgumentCollection - Processing 106433 bp from intervals
12:10:34.299 INFO  GenomicsDBImport - Done initializing engine
Created workspace /projects/karsanscratch/rdocking/KARSANBIO-1290_holiday_setup/A34430_small/work/joint/gatk-haplotype-joint/NA12878_small_batch/chr1_gl000191_random/NA12878_small_batch-chr1_gl000191_random_0_106433_genomicsdb
12:10:34.730 INFO  GenomicsDBImport - Vid Map JSON file will be written to NA12878_small_batch-chr1_gl000191_random_0_106433_genomicsdb/vidmap.json
12:10:34.730 INFO  GenomicsDBImport - Callset Map JSON file will be written to NA12878_small_batch-chr1_gl000191_random_0_106433_genomicsdb/callset.json
12:10:34.730 INFO  GenomicsDBImport - Complete VCF Header will be written to NA12878_small_batch-chr1_gl000191_random_0_106433_genomicsdb/vcfheader.vcf
12:10:34.730 INFO  GenomicsDBImport - Importing to array - NA12878_small_batch-chr1_gl000191_random_0_106433_genomicsdb/genomicsdb_array
12:10:34.738 INFO  ProgressMeter - Starting traversal
12:10:34.738 INFO  ProgressMeter -        Current Locus  Elapsed Minutes     Batches Processed   Batches/Minute
12:10:34.755 INFO  GenomicsDBImport - Importing batch 1 with 1 samples
#
# A fatal error has been detected by the Java Runtime Environment:
#
#  SIGSEGV (0xb) at pc=0x00007f7bab442c4d, pid=35911, tid=0x00007f7c18c65700
#
# JRE version: OpenJDK Runtime Environment (8.0_121-b15) (build 1.8.0_121-b15)
# Java VM: OpenJDK 64-Bit Server VM (25.121-b15 mixed mode linux-amd64 compressed oops)
# Problematic frame:
# C  [libtiledbgenomicsdb1654349342525822939.so+0x4fdc4d]  std::string::compare(char const*) const+0xd
#
# Failed to write core dump. Core dumps have been disabled. To enable core dumping, try "ulimit -c unlimited" before starting Java again
#
# An error report file with more information is saved as:
# /projects/karsanscratch/rdocking/KARSANBIO-1290_holiday_setup/A34430_small/work/joint/gatk-haplotype-joint/NA12878_small_batch/chr1_gl000191_random/hs_err_pid35911.log
#
# If you would like to submit a bug report, please visit:
#   http://www.azulsystems.com/support/
# The crash happened outside the Java Virtual Machine in native code.
# See problematic frame for where to report the bug.
#
' returned non-zero exit status 250
```

The command that was being run was:

```
[2018-01-31T20:10Z] unset JAVA_HOME && export PATH=/projects/rdocking_prj/software/bcbio-nextgen/data/anaconda/bin:$PATH && gatk-launch --java-options '-Xms727m -Xmx3272m -XX:+UseSerialGC -Djava.io.tmpdir=/projects/karsanscratch/rdocking/KARSANBIO-1290_holiday_setup/A34430_small/work/joint/gatk-haplotype-joint/NA12878_small_batch/chr1_gl000191_random/bcbiotx/tmptqq5o0' GenomicsDBImport --reader-threads 1 --genomicsdb-workspace-path NA12878_small_batch-chr1_gl000191_random_0_106433_genomicsdb -L chr1_gl000191_random:1-106433 --variant /projects/karsanscratch/rdocking/KARSANBIO-1290_holiday_setup/A34430_small/work/variation/rnaseq/gatk-haplotype/NA12878_small-gatk-haplotype.vcf.gz
```

I was trying to run RNA-seq variant calling using GATK4, using this version: `1.0.8a0` (updated from Github this morning)

Please let me know if there's any other details I can provide. Thanks!

**Comment by chapmanb** [comment]
Rod -- thanks for the report. This is a separate issue regarding GATK4 and GenomicsDB from the initial report. There was a bug in 4.0.0.0 that I reported to the GATK folks regarding GenomicsDB and long file paths. This should be resolved in 4.0.1.0 which we just updated to, so if you run an upgrade or `bcbio_conda install -c conda-forge -c bioconda gatk4` and re-run it should hopefully work for you.

**Comment by rdocking** [comment]
Great - thanks! I'll try this out and let you know if it doesn't resolve the issue.

**Comment by roryk** [comment]
Thanks, GATK has been updated a bunch since this issue was opened, so I think this is probably all set.
